jackson/src/main/scala/com/lightbend/lagom/internal/jackson/OldJacksonJsonSerializer.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package com.lightbend.lagom.internal.jackson
+
+import akka.actor.ExtendedActorSystem
+import akka.event.Logging
+import akka.serialization.BaseSerializer
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import akka.serialization.jackson.JacksonJsonSerializer
+import akka.serialization.jackson.JacksonMigration
+
+/**
+ * Placeholder for Lagom 1.5.x JacksonSerializer.
+ * Needed for deserialization of messages during rolling update, and old data (events, snapshots)
+ * that were serialized with the Lagom 1.5.x JacksonSerializer.
+ */
+private[lagom] class OldJacksonJsonSerializer(val system: ExtendedActorSystem)
+    extends SerializerWithStringManifest
+    with BaseSerializer {
+
+  private val log = Logging.getLogger(system, getClass)
+  private lazy val jacksonJsonSerializer: SerializerWithStringManifest = {
+    SerializationExtension(system)
+      .serializerByIdentity(JacksonJsonSerializer.Identifier)
+      .asInstanceOf[SerializerWithStringManifest]
+  }
+
+  if (system.settings.config.hasPath("lagom.serialization.json.migrations")) {
+    throw new IllegalStateException(
+      "JacksonJsonSerializer migrations defined in " +
+        s"'lagom.serialization.json.migrations' must be rewritten as [${classOf[JacksonMigration].getName}] " +
+        "and defined in config 'akka.serialization.jackson.migrations'."
+    )
+  }
+
+  override def manifest(obj: AnyRef): String =
+    jacksonJsonSerializer.manifest(obj)
+
+  override def toBinary(obj: AnyRef): Array[Byte] = {
+    log.warning(
+      "[{}] shouldn't be used for serialization of [{}]. Bind it to 'jackson-json' instead in " +
+        "'akka.actor.serialization-bindings' configuration.",
```

Comment:
I did some manual tests with rolling updates from 1.5.1 to local build of this branch and I found it annoying to have such a warning. 

When that happens, it means that the user opted-in for the old ser, so warning them about it seems not needed. I don't know... 🤔

Reply:
I think the problem with this warning is different: the warning is not helpful for Lagom users because in Lagom you use the `Jsonable` marker interface to make something serializable with `xxx-json` (old or new)

I agree there must be a warning but instead of being in every invocation of `toBinary` it should appear once at construction time.

Reply:
OldJacksonSerializer is gone and so is this warning. There is no need for it though, since `toBinary` will not be used unless user is doing some serious config overrides